src/infrastructure/server/routes/refresh/google.ts

```diff
@@ -0,0 +1,47 @@
+import Koa from 'koa'
+import Joi from '@hapi/joi'
+
+import { GoogleAuthService } from '../../../../infrastructure/services/GoogleAuthService'
+import { validateSchemaOrThrow } from '../../util'
+import { makeInvoker } from 'awilix-koa'
+
+interface RefreshBody {
+  refreshToken: string
+}
+
+interface Dependencies {
+  googleAuthService: GoogleAuthService
+}
+
+class GoogleRefreshApi {
+  private googleAuthService: GoogleAuthService
+
+  public constructor({ googleAuthService }: Dependencies) {
+    this.googleAuthService = googleAuthService
+  }
+
+  public refreshAccessToken: Koa.Middleware = async ctx => {
+    const body: RefreshBody = ctx.request.body
+    validateSchemaOrThrow(
+      Joi.object()
+        .keys({
+          refreshToken: Joi.string().required(),
+        })
+        .required(),
+      body
+    )
+
+    const accessToken = await this.googleAuthService.refreshAccessToken(
+      body.refreshToken
+    )
+
+    // eslint-disable-next-line no-param-reassign
```

Comment:
@francoischalifour I also like this eslint rule but it does not play well with Koa which forces you to mutate its context for responding to requests